lib/checkcondition.cpp

```diff
@@ -599,10 +590,10 @@ void CheckCondition::multiCondition2()
                                 tokens1.push(firstCondition->astOperand1());
                                 tokens1.push(firstCondition->astOperand2());
                             } else if (!firstCondition->hasKnownValue()) {
-                                if (isOppositeCond(false, mTokenizer->isCPP(), firstCondition, cond2, mSettings->library, true, true, &errorPath)) {
+                                if (isOppositeCond(false, mTokenizer->isCPP(), firstCondition, cond2, mSettings->library, true, true, &errorPath) && !isReturnVar) {
```

Comment:
I suggest that you check `isReturnVal` before the function call. That seems cheaper.